packages/env-tests/src/shared/attestation.ts

```diff
@@ -0,0 +1,256 @@
+    } else {
+      logger.info({ ...error }, 'Unknown error while revealing to issuer')
+    }
+  })
+}
+
+// Inefficient, but should be fine for now
+// Could mark already checked messages in the future
+async function findValidCode(
+  attestations: AttestationsWrapper,
+  messages: string[],
+  identifier: string,
+  attestationsToComplete: ActionableAttestation[],
+  account: string
+) {
+  for (const message of messages) {
+    try {
+      const code = AttestationUtils.extractAttestationCodeFromMessage(message)
+      if (!code) {
+        continue
+      }
+
+      const issuer = await attestations.findMatchingIssuer(
+        identifier,
+        account,
+        code,
+        attestationsToComplete.map((_) => _.issuer)
+      )
+
+      if (!issuer) {
+        continue
+      }
+
+      const isValid = await attestations.validateAttestationCode(identifier, account, issuer, code)
+
+      if (!isValid) {
+        continue
+      }
+
+      return { code, issuer }
+    } catch {
+      continue
+    }
+  }
+
+  return
+}
+
+export async function getPhoneNumber(twilioClient: Twilio, addressSid: string) {
+  const phoneNumber = await chooseFromAvailablePhoneNumbers(twilioClient)
+
+  if (phoneNumber !== undefined) {
+    return phoneNumber
+  }
+
+  return createPhoneNumber(twilioClient, addressSid)
+}
+
+async function chooseFromAvailablePhoneNumbers(twilioClient: Twilio) {
+  const availableNumbers = await twilioClient.incomingPhoneNumbers.list()
+  const usableNumber = availableNumbers[0]
+  return usableNumber
+}
+async function createPhoneNumber(twilioClient: Twilio, addressSid: string) {
+  const countryCodes = ['GB', 'US']
+  const countryCode = sample(countryCodes)
+  const context = await twilioClient.availablePhoneNumbers.get(countryCode!)
+  const numbers = await context.mobile.list({ limit: 10 })
+  const usableNumber = numbers[0]
+
+  await twilioClient.incomingPhoneNumbers.create({
+    phoneNumber: usableNumber!.phoneNumber,
+    addressSid,
+    // Just an requestbin.com endpoint to avoid errors
+    smsUrl: 'https://enzyutth0wxme.x.pipedream.net/',
+  })
+
+  return usableNumber
+}
+
+async function fetchLatestMessagesFromToday(client: Twilio, phoneNumber: string, count: number) {
+  return client.messages.list({
+    to: phoneNumber,
+    pageSize: count,
+    // Twilio keeps track of dates in UTC so it could be yesterday too
+    dateSentAfter: moment()
+      .subtract(2, 'day')
+      .toDate(),
+  })
+}
+
+export async function pollForMessagesAndCompleteAttestations(
+  attestations: AttestationsWrapper,
+  client: Twilio,
+  phoneNumber: string,
+  identifier: string,
+  account: Address,
+  attestationsToComplete: ActionableAttestation[],
+  txParams: CeloTransactionParams = {},
+  logger: Logger,
+  timeToPollForTextMessages: number
+) {
+  const startDate = moment()
+  logger.info({ pollingWait: POLLING_WAIT }, 'Poll for the attestation code')
```

Comment:
this is a constant log?

Reply:
Yeah, just to indicate to the log viewer what the constant value is